src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIKnight.java

```diff
@@ -164,6 +164,8 @@ protected IAIState attackPhysical()
         final IAIState state = preAttackChecks();
         if (state != getState())
         {
+            worker.getNavigator().clearPath();
+            worker.getMoveHelper().strafe(0, 0);
             setDelay(STANDARD_DELAY);
```

Comment:
nice find

Reply:
was working before, just moved it to a different place :D